extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -333,18 +348,54 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       dev().warn(TAG, `Frame already exists, sra: ${this.useSra}`);
       return '';
     }
+
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    let rtcRequestPromise = null;
+    let rtcConfig;
+    if (ampRtcPageElement) {
+      rtcConfig = tryParseJson(ampRtcPageElement.innerHTML);
+      if (rtcConfig && typeof rtcConfig == 'object'
+      && !!rtcConfig['doubleclick']) {
+        rtcRequestPromise = this.sendRtcRequestPromise_(/** @type {!Object} */(
+            rtcConfig['doubleclick']));
+      }
+    }
     // TODO(keithwrightbos): SRA blocks currently unnecessarily generate full
     // ad url.  This could be optimized however non-SRA ad url is required to
     // fallback to non-SRA if single block.
     this.populateAdUrlState();
     // TODO: Check for required and allowed parameters. Probably use
     // validateData, from 3p/3p/js, after noving it someplace common.
     const startTime = Date.now();
-    return getPageLevelParameters_(this.win, this.getAmpDoc(), startTime)
-        .then(pageLevelParameters =>
-        googleAdUrl(this, DOUBLECLICK_BASE_URL, startTime,
-            Object.assign(this.getBlockParameters_(), pageLevelParameters),
-          ['108809080']));
+    const pageLevelParametersPromise = getPageLevelParameters_(
```

Comment:
RTC modifies block level parameters so you don't actually have to wait on page level before merging.  You can move the merge directly into sendRTCRequest.  This has the added benefit of allowing you to add other parameters (e.g. time to get RTC response) without putting that logic in getAdUrl